### PR TITLE
Add more complete type hints for pkg_resources.

### PR DESCRIPTION
**Pull request by Solumin**
The current file only covers `pkg_resources/__init__.py`.
`pkg_resources/py31compat.pyi` is used by setuptools, for example.

### CODE REVIEW

File: third_party/3/pkg_resources/py31compat.pyi [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import os
+import sys
+
+needs_makedirs: bool
+
+def _makedirs_31(path: Text, exist_ok: bool = ...) -> None: ...
+
+# _makedirs_31 has special behavior to handle an edge case that was removed in
+# 3.4.1. No one should be using 3.4 instead of 3.4.1, so this should be fine.
+if sys.version_info >= (3, 4):
```
Comment (rchen152) [inline third_party/3/pkg_resources/py31compat.pyi]: `py31compat.py` has:
```
needs_makedirs = (
    six.PY2 or
    (3, 4) <= sys.version_info < (3, 4, 1)
)
makedirs = _makedirs_31 if needs_makedirs else os.makedirs
```
(https://github.com/pypa/setuptools/blob/master/pkg_resources/py31compat.py)
so `_makedirs_31` is used in Python 2 and Python 3.4.0. If we go with your assumption that we can ignore 3.4.0 (which I think is fine), then the version check should be `>= (3,)`.

Comment (Solumin) [reply third_party/3/pkg_resources/py31compat.pyi]: Done.

**Comment by Solumin** [comment]
```
E           AssertionError: usage: mypy [-h] [-v] [-V] [more options; see below]
E                         [-m MODULE] [-p PACKAGE] [-c PROGRAM_TEXT] [files ...]
E             mypy: error: Python version (3, 6) did not match executable /tmp/tmpvyemosai/bin/python, got version (2, 7).
E
E           assert '' == "/tmp/tmpykgf5e0l/simple.py:...ltins.list[builtins.str]'\n"
E             + /tmp/tmpykgf5e0l/simple.py:3: error: Module 'typedpkg' has no attribute 'dne'
E             + /tmp/tmpykgf5e0l/simple.py:5: error: Revealed type is 'builtins.list[builtins.str]'
```
This failure doesn't look related to my changes.

**Comment by srittau** [comment]
I have restarted the build job.

**Pull request merged**